src/test/java/org/sagebionetworks/bridge/spring/controllers/ParticipantDataControllerTest.java

```diff
@@ -0,0 +1,324 @@
+                .withConsented(true).withRequired(true).withSignedMostRecentConsent(true).build();
+        Map<SubpopulationGuid, ConsentStatus> statuses = Maps.newHashMap();
+        statuses.put(SubpopulationGuid.create(status.getSubpopulationGuid()), status);
+
+        session = new UserSession(participant);
+        session.setAppId(TEST_APP_ID);
+        session.setAuthenticated(true);
+        session.setConsentStatuses(statuses);
+
+        participantData = createParticipantData("a", "b");
+
+        doReturn(app).when(mockAppService).getApp(TEST_APP_ID);
+        doReturn(TEST_USER_ID).when(mockAccount).getId();
+        doReturn(session).when(controller).getSessionIfItExists();
+        doReturn(session).when(controller).getAuthenticatedSession(ADMIN, WORKER);
+        doReturn(session).when(controller).getAuthenticatedAndConsentedSession();
+
+        doReturn(mockRequest).when(controller).request();
+    }
+
+    @Test
+    public void verifyAnnotations() throws Exception {
+        assertCrossOrigin(ParticipantDataController.class);
+        assertGet(ParticipantDataController.class, "getAllDataForUser");
+        assertGet(ParticipantDataController.class, "getDataByIdentifierForSelf");
+        assertGet(ParticipantDataController.class, "getAllDataForAdminWorker");
+        assertGet(ParticipantDataController.class, "getDataByIdentifierForAdminWorker");
+        assertCreate(ParticipantDataController.class, "saveDataForSelf");
+        assertCreate(ParticipantDataController.class, "saveDataForAdminWorker");
+        assertDelete(ParticipantDataController.class, "deleteDataByIdentifier");
+        assertDelete(ParticipantDataController.class, "deleteDataByIdentifier");
+        assertDelete(ParticipantDataController.class, "deleteDataForAdmin");
+    }
+
+    @Test
+    public void testGetAllDataForSelf() {
+        doReturn(makeResults(OFFSET_KEY, PAGE_SIZE_INT)).when(mockParticipantDataService)
+                .getAllParticipantData(TEST_USER_ID, OFFSET_KEY, PAGE_SIZE_INT);
+
+        ForwardCursorPagedResourceList<String> result = controller.getAllDataForSelf(OFFSET_KEY, PAGE_SIZE_STRING);
+
+        assertEquals(NEXT_PAGE_OFFSET_KEY, result.getNextPageOffsetKey());
+        assertEquals(result.getItems().get(0), IDENTIFIER);
+        assertEquals(result.getItems().get(1), IDENTIFIER);
+    }
+
+    @Test
+    public void testGetDataByIdentifierForSelf() {
+        doReturn(participantData).when(mockParticipantDataService).getParticipantData(session.getId(), IDENTIFIER);
+
+        ParticipantData result = controller.getDataByIdentifierForSelf(IDENTIFIER);
+
+        assertEquals(result.getUserId(), participantData.getUserId());
+        assertEquals(result.getIdentifier(), participantData.getIdentifier());
+        assertEquals(result.getData(), participantData.getData());
+        assertEquals(result.getVersion(), participantData.getVersion());
+        assertSame(result, participantData);
```

Comment:
If you verify that result and participantData are the same, you don't need to verify that all the parameters are equal, because they are guaranteed to be equal.